### ISSUE TITLE
`nosetests test_poly` isn't thread safe

### ISSUE THREAD

**Issue by jmeyers314** [rmjarvis/Piff#41]
A few times every ten attempts or so, I'm getting errors when running `nosetests test_poly.py` from the test directory.  Looks like `test_poly_load_save` and `test_poly_load_err` are both calling `poly_load_save_sub` on different threads, leading to problems.

**Comment by gogrean** [comment]
Did you get an error similar to the one at the end of my post? The only time the test failed for me was when running it on multiple cores. I added a filename option to `poly_load_save_sub` (PR #46), which fixes the error below.

```
Traceback (most recent call last):
  File "/mnt/hd/gogrean/anaconda3/envs/piffpy2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/mnt/hd/gogrean/code/Piff/tests/test_poly.py", line 381, in test_poly_load_save
    poly_load_save_sub(poly_type,poly_type)
  File "/mnt/hd/gogrean/code/Piff/tests/test_poly.py", line 336, in poly_load_save_sub
    with fitsio.FITS(filename,'rw',clobber=True) as f:
  File "/mnt/hd/gogrean/anaconda3/envs/piffpy2/lib/python2.7/site-packages/fitsio/fitslib.py", line 353, in __init__
    self._FITS =  _fitsio_wrap.FITS(filename, self.intmode, create)
IOError: FITSIO status = 105: couldn't create the named file
failed to create new file (already exists?):
output/poly_test_file.fits
```

**Comment by jmeyers314** [comment]
Hi Georgiana,

Yes.  I think this fixes my issue.  Thanks!

**Comment by rmjarvis** [comment]
Merged via PR #46
